src/dl.cpp

```diff
@@ -247,18 +247,78 @@ static dl_error_t dl_internal_store_array( dl_ctx_t dl_ctx, dl_type_storage_t st
+				if( DL_ERROR_OK != err )
+					return err;
+				dl_binary_writer_write_uint32( &store_ctx->writer, tid );
+				if (sizeof(uintptr_t) == 8)
+					dl_binary_writer_write_zero( &store_ctx->writer, 4 );
+			}
+			break;
 		default: // default is a standard pod-type
 			dl_binary_writer_write( &store_ctx->writer, instance, count * size );
 			break;
 	}
 	return DL_ERROR_OK;
 }
 
-static dl_error_t dl_internal_store_member( dl_ctx_t dl_ctx, const dl_member_desc* member, uint8_t* instance, CDLBinStoreContext* store_ctx )
+static dl_error_t dl_internal_store_anyarray( dl_ctx_t dl_ctx, const uint8_t* data_ptr, uint32_t count, CDLBinStoreContext* store_ctx )
+{
+	for (uint32_t i = 0; i < count; ++i)
+	{
+		uintptr_t sub_pos = dl_binary_writer_tell( &store_ctx->writer );
+		dl_binary_writer_seek_end( &store_ctx->writer );
+
+		const uint8_t* sub_data_ptr = data_ptr + i * 3 * sizeof(void*);
+		const uint8_t* sub_data = *(const uint8_t* const*)sub_data_ptr;
+		uint32_t sub_count  = *(const uint32_t*)( sub_data_ptr + sizeof(void*) );
+		dl_typeid_t sub_type_id = *(const dl_typeid_t*)( sub_data_ptr + 2 * sizeof(void*) );
+		const dl_type_desc* sub_type = dl_internal_find_type( dl_ctx, sub_type_id );
+		size_t sub_size = dl_internal_align_up( sub_type->size[DL_PTR_SIZE_HOST], sub_type->alignment[DL_PTR_SIZE_HOST] );
+		dl_binary_writer_align( &store_ctx->writer, sub_type->alignment[DL_PTR_SIZE_HOST] );
+
+		uintptr_t sub_offset = dl_binary_writer_tell( &store_ctx->writer );
+
+		// write data!
+		dl_binary_writer_reserve( &store_ctx->writer, sub_count * sub_size ); // reserve space for array so subdata is placed correctly
+
+		dl_error_t err = dl_internal_store_array(dl_ctx, DL_TYPE_STORAGE_STRUCT, sub_type, sub_data, sub_count, sub_size, store_ctx);
+		if (DL_ERROR_OK != err)
+			return err;
+		dl_binary_writer_seek_set( &store_ctx->writer, sub_pos );
+								
+		// use when fast_loading is merged 
+		//if( !store_ctx->writer.dummy )
+		//	store_ctx->ptrs.Add( dl_binary_writer_tell( &store_ctx->writer ) );
+
+		// make room for ptr
+		dl_binary_writer_write( &store_ctx->writer, &sub_offset, sizeof(uintptr_t) );
+
+		// write count
+		dl_binary_writer_write( &store_ctx->writer, &sub_count, sizeof(uint32_t) );
+		if (sizeof(uintptr_t) == 8)
+			dl_binary_writer_write_zero( &store_ctx->writer, sizeof(uint32_t) );
+
+		// write type id
+		dl_binary_writer_write( &store_ctx->writer, &sub_type_id, sizeof(uint32_t) );
+		if (sizeof(uintptr_t) == 8)
+			dl_binary_writer_write_zero( &store_ctx->writer, sizeof(uint32_t) );
+	}
+	return DL_ERROR_OK;
+}
+
+dl_error_t dl_internal_store_member( dl_ctx_t dl_ctx, const dl_member_desc* member, const uint8_t* instance, CDLBinStoreContext* store_ctx )
 {
 	dl_type_atom_t    atom_type    = member->AtomType();
 	dl_type_storage_t storage_type = member->StorageType();
 
+	if (storage_type == DL_TYPE_STORAGE_ANYARRAY)
+		atom_type = DL_TYPE_ATOM_ARRAY; // To avoid code duplification and simplify implementation
```

Comment:
:point_up:

Reply:
Would you prefer the code duplification here?